File: 781.rabbits-in-forest.go
/*
 * @lc app=leetcode id=781 lang=golang
 *
 * [781] Rabbits in Forest
 */

// @lc code=start
func numRabbits(answers []int) int {
	total, m := 0, make(map[int]int)
    
	for _, v := range answers{
		if m[v] == 0{
			m[v] = v
			total += v + 1
		}else{
			m[v]--
		}
	}

	return total
}
// @lc code=end

File: 781.rabbits-in-forest.cpp
/*
 * @lc app=leetcode id=781 lang=cpp
 *
 * [781] Rabbits in Forest
 */

// @lc code=start
class Solution {
 public:
  int numRabbits(vector<int>& answers) {
    unordered_map<int, int> m;

    int total = 0;
    for (int ans : answers) {
      if (m[ans] == 0) {
        m[ans] = ans;
        total += ans + 1;
      } else {
        m[ans] -= 1;
      }
    }

    return total;
  }
};
// @lc code=end
